torch/ao/quantization/quantize_pt2e.py

```diff
@@ -66,6 +69,23 @@ def prepare_qat_pt2e(
     model = _disallow_eval_train(model)
     return model
 
+_QUANT_OPS = [
+    torch.ops.quantized_decomposed.quantize_per_tensor.default,
+    torch.ops.quantized_decomposed.quantize_per_tensor.tensor,
+    torch.ops.quantized_decomposed.quantize_per_channel.default,
+]
+def _quant_node_filter(n: Node) -> bool:
+    """If there is any pure ops between get_attr and quantize op they will be const propagated
+    e.g. get_attr(weight) -> transpose -> quantize -> dequantize*
+    (Note: dequantize op is not going to be constant propagated)
+
+    Note: this filter does not handle the case when there are any additional ops after quantize:
+    e.g. get_attr(weight) -> transpose1 -> quantize -> transpose2 -> dequantize*
+    this filter will filter out the above example since the last op in the sequence and we'll not
```

Comment:
This is because the const prop will be associated with "tranpose" node, for which this function will return false? Makes sense, but just trying to understand.

Reply:
yes, in this case when we replace node with constant Tensors, we'll be replacing transpose2 node with a constant Tensor and the check will fail

Reply:
May I know the motivation of this filtering? What's the problem if we fold the weights till `transpose2`?

Reply:
this is not a good example I think, the point is that we only fold things related to q/dq, not folding `transpose2` is an artifact of the current implementation details, it's not that important.